### DISCUSSION TITLE
Cloud storage

### DISCUSSION THREAD

**Discussion by Suncatcher** [MusicPlayerDaemon/MPD#2078] [Ideas]
Hi, 
does MPD support playing music files from the cloud providers (OneDrive, DropBox, Google Drive etc)? Briefly searched and didn't find any questions like this.
Also checked the documentation chapter about [The music directory and the database](https://mpd.readthedocs.io/en/stable/user.html#the-music-directory-and-the-database), no mentions about storages other than local.

**Comment by MaxKellermann** [comment]
No, it doesn't. I have no account in any cloud storage, and thus have no interest in developing such a feature. But if code were submitted to me, it would probably be merged.

**Comment by Suncatcher** [comment]
Ok, got it. Unfortunately I have no skills in C++, so cannot contribute it here. Let's hope somebody interested in that will be able to introduce this feature.

**Comment by skidoo23** [comment]
In my humble opinion, access to commercial cloud storage providers is outside the scope of this project. Luckily for you, there are already solutions for integrating it as a file system: for example onedriver, dbxfs or s3fs. The MPD project would certainly be happy to receive configuration examples for various cloud storage providers.

**Comment by Suncatcher** [comment]
In the manual of the MPD I see passage `"Through plugins and libraries it can play a variety of sound files while being controlled by its network protocol"` so I assume the MPD was created and developed with an idea of modular structure, so it is inclined to be extensible through plugins, which should be embraced, not discouraged.

> In my humble opinion, access to commercial cloud storage providers is outside the scope of this project

I am respectfully disagree with that statement. What is in scope and what is out of scope is defined purely and exclusively by the project owner philosophy and/or its maintainers. In my opinion having different storage options fully fits modular philosophy of that project.

Also, worth mentioning that word `commercial cloud storage providers` has no practical sense, as all cloud providers are commercial, unless they are phishing users credentials and/or personal data providing storage "for free". Having hosted my own NAS storage with OwnCloud framework as a web-access, I can not name myself as a *provider*, because I used it only for my needs.
